apps/files/models.py

```diff
@@ -581,6 +581,8 @@ class FileUpload(amo.models.ModelBase):
         choices=amo.APPS_CHOICES, db_column="compat_with_app_id", null=True)
     compat_with_appver = models.ForeignKey(
         AppVersion, null=True, related_name='uploads_compat_for_appver')
+    version = models.CharField(max_length=255, null=True)
+    addon = models.ForeignKey('addons.Addon', null=True)
```

Comment:
why add `version` and `addon`? A valid `FileUpload` always results in a `File`, right? Couldn't you just link it with a `file_id` to `File` instead? That way you could get `file_upload.file.version.addon ...`